polaris-ui/scripts/build.sh

```diff
@@ -18,6 +18,6 @@
 #   do.
 rm -rf build
 PUBLIC_URL=/polaris-ui BUILD_PATH=build/polaris-ui react-scripts build
-cp -r build/polaris-ui/static/js build/polaris-ui/static/js-pre-substitution
 mv build/polaris-ui/index.html build/index.html
-mv build/polaris-ui/run-substitution.sh build/run-substitution.sh
\ No newline at end of file
+cp env-variable-substitution/substitute-config.js build/substitute-config.js
```

Comment:
Rather than keep `substitute-config.js` in the `public` folder, keep it a dedicated `env-variable-substitution` folder together with unit tests.  After the build, copy the file to the output folder so it is available as before.